src/ensembl/src/content/app/browser/browser-region-field/BrowserRegionField.tsx

```diff
@@ -0,0 +1,169 @@
+import { RootState } from 'src/store';
+import {
+  changeBrowserLocation,
+  changeFocusObject,
+  toggleRegionFieldActive
+} from '../browserActions';
+import {
+  getBrowserActiveGenomeId,
+  getRegionEditorActive,
+  getRegionFieldActive,
+  getChrLocation
+} from '../browserSelectors';
+import { getChrLocationFromStr, validateRegion } from '../browserHelper';
+
+import applyIcon from 'static/img/shared/apply.svg';
+import clearIcon from 'static/img/shared/clear.svg';
+
+import styles from './BrowserRegionField.scss';
+import browserStyles from '../Browser.scss';
+import browserNavBarStyles from '../browser-nav/BrowserNavBar.scss';
+
+export type BrowserRegionFieldProps = {
+  activeGenomeId: string | null;
+  chrLocation: ChrLocation | null;
+  isActive: boolean;
+  isDisabled: boolean;
+  changeBrowserLocation: (genomeId: string, chrLocation: ChrLocation) => void;
+  changeFocusObject: (objectId: string) => void;
+  toggleRegionFieldActive: (regionFieldActive: boolean) => void;
+};
+
+export const BrowserRegionField = (props: BrowserRegionFieldProps) => {
+  const { activeGenomeId, chrLocation } = props;
+  const [regionFieldInput, setRegionFieldInput] = useState('');
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const activateForm = () => {
+    if (!props.isDisabled) {
+      props.toggleRegionFieldActive(true);
+    }
+  };
+
+  const changeRegionFieldInput = (value: string) => setRegionFieldInput(value);
+
+  const getRegionInputWithRegion = (input: string) => {
+    const [stick] = chrLocation as ChrLocation;
+
+    return input.includes(':') ? input : `${stick}:${input}`;
+  };
+
+  const resetForm = () => {
+    setRegionFieldInput('');
+    setErrorMessage(null);
+    props.toggleRegionFieldActive(false);
+  };
+
+  const onValidationError = (errorMessages: any) => {
+    const { regionError, startError, endError } = errorMessages;
+    const errorMessageToBeDisplayed = regionError || startError || endError;
+    setErrorMessage(errorMessageToBeDisplayed);
+  };
+
+  const onValidationSuccess = (regionId: string) => {
+    resetForm();
+
+    const [stick] = chrLocation as ChrLocation;
+    const stickInput = regionFieldInput.includes(':')
+      ? regionFieldInput.split(':')[0]
+      : stick;
+
+    if (stickInput === stick) {
+      props.changeBrowserLocation(
+        props.activeGenomeId as string,
+        getChrLocationFromStr(getRegionInputWithRegion(regionFieldInput))
+      );
+    } else {
+      props.changeFocusObject(regionId);
+    }
+  };
+
+  const closeForm = () => resetForm();
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (activeGenomeId && regionFieldInput && chrLocation) {
+      setErrorMessage(null);
+
+      validateRegion({
+        regionInput: getRegionInputWithRegion(regionFieldInput),
+        genomeId: props.activeGenomeId,
+        onSuccess: onValidationSuccess,
+        onError: onValidationError
+      });
```

Comment:
just curious why you decided to use this interface for `validateRegion` instead of having is an async function?

Reply:
As discussed offline, I'll leave the code as it is for now.